vllm/datadog_checks/vllm/check.py

```diff
@@ -0,0 +1,94 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import AgentCheck, OpenMetricsBaseCheckV2  # noqa: F401
+
+from .metrics import METRIC_MAP, RENAME_LABELS_MAP
+
+DEFAULT_VERSION_ENDPOINT = "/version"
+DEFAULT_HEALTH_ENDPOINT = "/health"
+
+
+class vLLMCheck(OpenMetricsBaseCheckV2):
+
+    # This will be the prefix of every metric and service check the integration sends
+    DEFAULT_METRIC_LIMIT = 0
+    __NAMESPACE__ = 'vllm'
+
+    def __init__(self, name, init_config, instances):
+        super(vLLMCheck, self).__init__(name, init_config, instances)
+
+        self.openmetrics_endpoint = self.instance.get("openmetrics_endpoint")
+        self.tags = self.instance.get('tags', [])
+        self.version_endpoint = DEFAULT_VERSION_ENDPOINT
+        self._health_endpoint = DEFAULT_HEALTH_ENDPOINT
+
+        # Get the API server port if specified, otherwise use the default 8000.
+        self.server_port = str(self.instance.get('server_port', 8000))
+
+        self.collect_server_info = self.instance.get('collect_server_info', True)
+
+        # Get the base url from the openmetrics endpoint and construct the server info API endpoint.
+        if self.collect_server_info and self.openmetrics_endpoint:
+            parts = urlparse(self.openmetrics_endpoint)
+            # Delete the /metrics from the url
+            self.base_url = parts._replace(path="")
+            # Replace the openmetrics port by the server port
+            self.server_info_api = self.base_url._replace(netloc=parts.hostname + ':' + self.server_port).geturl()
+        else:
+            self.log.debug("Collecting server info through API is disabled.")
+            return
+
+    def get_default_config(self):
+        return {
+            'metrics': [METRIC_MAP],
+            "rename_labels": RENAME_LABELS_MAP,
+        }
+
+    @AgentCheck.metadata_entrypoint
+    def _submit_version_metadata(self):
+
+        endpoint = self.instance.get("openmetrics_endpoint").replace("/metrics", DEFAULT_VERSION_ENDPOINT)
```

Comment:
```suggestion
        endpoint = self.instance.["openmetrics_endpoint"].replace("/metrics", DEFAULT_VERSION_ENDPOINT)
```
`openmetrics_endpoint` is a required field, we can assume it's present.